connman/src/service.c

```diff
@@ -8153,16 +8281,26 @@ int __connman_service_online_check_failed(struct connman_service *service,
 		;
 	}
 
-	if (*online_check_count > 1)
-		--(*online_check_count);
-	else
-		connman_warn("Online check failed for %p %s", service, service->name);
+	timeout = connman_wakeup_timer_add_seconds(*interval * *interval,
+				redo_func, connman_service_ref(service));
+	if (type == CONNMAN_IPCONFIG_TYPE_IPV4) {
+		if (service->online_timeout_ipv4)
+			g_source_remove(service->online_timeout_ipv4);
```

Comment:
does not clear service->online_timeout_ipv4?
ditto for ipv6 timeout below

Reply:
Wouldn't that be kind of moot as it is being setup with the new timeout few lines after?

Reply:
hmm, yes, my pattern matching misfired -> is ok as-is